_rules/required-form-fields-not-completed-334972.md

```diff
@@ -0,0 +1,243 @@
+---
+id: 334972
+name: Error message identifies required fields left unfilled
+rule_type: atomic
+description: |
+  This rule checks that text error messages identify required form fields that were not completed.
+accessibility_requirements:
+  wcag20:3.3.1: # Error Identification (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+  wcag-technique:G83: # Providing text descriptions to identify required fields that were not completed
+    forConformance: false
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree # (e.g. HTTP Messages, DOM Tree, CSS Styling, Accessibility Tree, Language, etc.,)
+  - CSS Styling
+  - Language
+acknowledgements:
+  authors:
+    - Carlos Duarte
+    - João Vicente
+---
+
+## Applicability
+
+The rule applies to each [required][] HTML element that:
```

Comment:
Browsers automatically validate form fields that have the `required` attribute. It seems to me that the only way to fail this rule is to run it in a browser that doesn't support the `required` attribute, or to put the `required` attribute on an element that doesn't support it (like `<div required role="textbox"></div>`). I don't really see how this is a useful rule, unless maybe you make this about aria-required=true?

Reply:
But I'm not just using the `required` attribute in the definition of required. The goal was to have the rule also consider those instances where required fields are not marked with the `required` attribute.